app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/SignatureVerifier.java

```diff
@@ -0,0 +1,78 @@
+package com.appsmith.server.helpers;
+
+import com.appsmith.server.exceptions.AppsmithError;
+import com.appsmith.server.exceptions.AppsmithException;
+import org.apache.commons.collections.CollectionUtils;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.crypto.params.AsymmetricKeyParameter;
+import org.bouncycastle.crypto.util.OpenSSHPublicKeyUtil;
+import org.pf4j.util.StringUtils;
+import org.springframework.http.HttpHeaders;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Objects;
+
+import static com.appsmith.server.constants.ApiConstants.APPSMITH_SIGNATURE;
+import static com.appsmith.server.constants.ApiConstants.DATE;
+
+public class SignatureVerifier {
+    private static final String publicVerificationKey = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICNwJ+zx2opXjjOga/YyzRxb2czvNgQ/twA+miCKDIX3 appsmith";
+
+    private static final String TIMESTAMP = "timestamp";
+
+    private static final String EQUAL = "=";
+
+    /**
+     * Method to verify the API signature from CS.
+     * @param headers   Response headers from CS
+     * @return          If the signature is valid
+     */
+    public static boolean isSignatureValid(HttpHeaders headers) {
+        if (CollectionUtils.isEmpty(headers.get(APPSMITH_SIGNATURE))) {
+            return false;
+        }
+        String signature = Objects.requireNonNull(headers.get(APPSMITH_SIGNATURE)).get(0);
```

Comment:
The `isEmpty` above will do null-check too, so we can remove the `requireNonNull` call here.

```suggestion
        String signature = headers.get(APPSMITH_SIGNATURE).get(0);
```

Reply:
Will remove this I was of the same opinion but IntelliJ started throwing warning hence kept it.